django_guid/middleware.py

```diff
@@ -22,95 +25,96 @@ def __init__(self, get_response):
+        :return: None
         """
         cls._guid.pop(threading.current_thread(), None)
 
     @staticmethod
     def _generate_guid() -> str:
         """
-        Generates an UUIDv4/GUID as a string
+        Generates an UUIDv4/GUID as a string.
         :return: GUID
         """
         return uuid.uuid4().hex
 
     @staticmethod
     def _validate_guid(original_guid: str) -> bool:
         """
-        Validates a GUID
-        :param original_guid: GUID to check
+        Validates a GUID.
+        :param original_guid: string to validate
         :return: bool
         """
         try:
-            guid_value = uuid.UUID(original_guid, version=4)
+            return original_guid == uuid.UUID(original_guid, version=4).hex
         except ValueError:
             return False
-        return original_guid == guid_value.hex
 
     def _get_correlation_id_from_header(self, request: HttpRequest) -> str:
         """
         Returns either the provided GUID or a new one,
-        depending on if it's a valid GUID or not and the specified settings
+        depending on if the provided GUID is valid, and the specified settings
         :param request: HttpRequest object
         :return: GUID
         """
-        given_guid = str(request.headers.get('Correlation-ID'))
-        if settings.VALIDATE_GUID is True and self._validate_guid(given_guid):
+        given_guid = str(request.headers.get(settings.GUID_HEADER_NAME))
+        if not settings.VALIDATE_GUID:
             return given_guid
-        elif settings.VALIDATE_GUID is False:
+        elif settings.VALIDATE_GUID and self._validate_guid(given_guid):
             return given_guid
-        return self._generate_guid()
+        else:
+            return self._generate_guid()
 
     def _get_id_from_header(self, request: HttpRequest) -> str:
         """
-        Checks if there is a header with the specified name. Default is `Correlation-ID`.
-        If there is, it will fetch it and potentially validate it as a GUID, based on the settings.
-        
-        If there is no header found, it will generate a GUID.
-        :param request: Request object
+        Checks if the request contains the header specified in the Django settings.
+        If there is, we fetch the header and attempt to validate the contents as GUID, based on the settings.
+        If no header is found, we generate a GUID to be injected instead.
+        :param request: HttpRequest object
         :return: GUID
         """
         guid_header_name = settings.GUID_HEADER_NAME
+
         if request.headers.get(guid_header_name):  # Case insensitive headers.get added in Django2.2 so this is safe
             logger.info(f'{guid_header_name} found in the header: {request.headers.get(guid_header_name)}')
-            request.correlation_id = self._get_correlation_id_from_header(request)
+            return self._get_correlation_id_from_header(request)
```

Comment:
This is changing workflow. I'm adding the `correlation_id` to the `request` object. In other words I'm making the `request.correlation_id` variable accessible in later middleware and views. This would break that.

Reply:
Fixed! Reverted back to the way it was before 👍